### DISCUSSION TITLE
Facing Web3Provider error

### DISCUSSION THREAD

**Discussion by SamsShow** [RequestNetwork/requestNetwork#1515] [Questions] [answered]
Failed to initialize Request Network: TypeError: Cannot read properties of undefined (reading 'Web3Provider')

async initialize() {
    try {
      if (!window.ethereum) {
        throw new Error('No Web3 provider found. Please install MetaMask.');
      }

      const provider = new ethers.providers.Web3Provider(window.ethereum);
      await provider.send("eth_requestAccounts", []);
      
      const signer = provider.getSigner();
      
      this.signatureProvider = new Web3SignatureProvider({
        provider: window.ethereum,
        signer
      });
      
      this.requestClient = new RequestNetwork({
        nodeConnectionConfig: {
          baseURL: 'https://sepolia.gateway.request.network/',
        },
        signatureProvider: this.signatureProvider,
      });

      return this.requestClient;
    } catch (error) {
      console.error('Failed to initialize Request Network:', error);
      throw error;
    }
  }

**Comment by aimen74** [comment, marked as answer]
@SamsShow, you need to use the provider that you got from others when creating a new `Web3SignatureProvider`, not the `window. ethereum` directly.

Example:

```typescript
const   ethersProvider = new ethers.providers.Web3Provider(walletProvider);


 const web3SignatureProvider = new Web3SignatureProvider(
    ethersProvider!.provider
  );
  ```

In your case it would be 
```typescript
 const provider = new ethers.providers.Web3Provider(window.ethereum);
  await provider.send("eth_requestAccounts", []);
  
  const signer = provider.getSigner();
  
  // Fixed here
  this.signatureProvider = new Web3SignatureProvider(provider.provider);
  
  this.requestClient = new RequestNetwork({
    nodeConnectionConfig: {
      baseURL: 'https://sepolia.gateway.request.network/',
    },
    signatureProvider: this.signatureProvider,
  });

  return this.requestClient;
} catch (error) {
  console.error('Failed to initialize Request Network:', error);
  throw error;
}
```
